hackathon_site/dashboard/frontend/src/slices/hardware/cartSlice.test.ts

```diff
@@ -107,3 +107,23 @@ describe("removeFromCart action", () => {
         });
     });
 });
+
+describe("updateCart action", () => {
+    test("updateCart updates quantity of existing item", async () => {
+        const store = makeStoreWithEntities({
+            cartItems: mockCartItems,
+        });
+
+        store.dispatch(
+            updateCart({ id: mockCartItems[0].hardware_id, changes: { quantity: 25 } })
+        );
+
+        await waitFor(() => {
+            expect(cartSelectors.selectAll(store.getState()).length).toEqual(3);
```

Comment:
```suggestion
            expect(cartSelectors.selectAll(store.getState()).length).toEqual(mockCartItems.length);
```

Just in case we add more items to that array